modules/porous_flow/src/materials/PorousFlowFluidStateWaterNCG.C

```diff
@@ -226,35 +226,28 @@ PorousFlowFluidStateWaterNCG::thermophysicalProperties() const
   }
 
   // Set the phase densities and viscosities
-  _fluid_density[_qp][_aqueous_phase_number] = liquid_density;
-  _fluid_density[_qp][_gas_phase_number] = gas_density;
-
-  _fluid_viscosity[_qp][_aqueous_phase_number] = liquid_viscosity;
-  _fluid_viscosity[_qp][_gas_phase_number] = gas_viscosity;
+  fsp.fluid_density[_aqueous_phase_number] = liquid_density;
+  fsp.fluid_density[_gas_phase_number] = gas_density;
+  fsp.fluid_viscosity[_aqueous_phase_number] = liquid_viscosity;
+  fsp.fluid_viscosity[_gas_phase_number] = gas_viscosity;
 
   // Derivatives wrt PorousFlow variables are required by the kernels
   // Note: these don't need to be stateful so don't calculate them in
   // initQpStatefulProperties
   if (!_is_initqp)
```

Comment:
And by `if (!_is_nodal)` section, I meant the `if (!_nodal_material)` that is later in this function at the same scope as the `if (!_is_initqp)`.  Apologies for my initial, bug-ridden comment.